spec/models/validations/household_validations_spec.rb

```diff
@@ -307,33 +309,87 @@
+        Timecop.return
+        Singleton.__init__(FormHandler)
+      end
+
+      context "when the household contains a person under 16" do
+        it "validates that person must be a child of the tenant" do
+          record.age2 = 14
+          record.relat2 = "P"
+          household_validator.validate_person_age_matches_relationship(record)
+          expect(record.errors["relat2"])
+            .to include(match I18n.t("validations.household.relat.child_under_16_lettings", person_num: 2))
+          expect(record.errors["age2"])
+            .to include(match I18n.t("validations.household.age.child_under_16_relat_lettings", person_num: 2))
+        end
+
+        it "expects that person is a child of the tenant" do
+          record.age2 = 14
+          record.relat2 = "C"
+          household_validator.validate_person_age_matches_relationship(record)
+          expect(record.errors["relat2"]).to be_empty
+          expect(record.errors["age2"]).to be_empty
+        end
+      end
+    end
+
+    context "with 2024 logs" do
+      let(:log_date) { Time.zone.local(2024, 4, 1) }
+
+      before do
+        Timecop.freeze(log_date)
+        Singleton.__init__(FormHandler)
+      end
+
+      after do
+        Timecop.return
+        Singleton.__init__(FormHandler)
+      end
+
+      it "validates person under 16 is not partner" do
         record.age2 = 14
         record.relat2 = "P"
-        household_validator.validate_household_number_of_other_members(record)
+        household_validator.validate_person_age_matches_relationship(record)
         expect(record.errors["relat2"])
-          .to include(match I18n.t("validations.household.relat.child_under_16_lettings", person_num: 2))
+          .to include("Answer cannot be ‘partner’ as you told us person 2's age is under 16")
         expect(record.errors["age2"])
-          .to include(match I18n.t("validations.household.age.child_under_16_relat_lettings", person_num: 2))
+          .not_to include("Age cannot be under 16 as you told us person 2's relationship to the lead tenant is partner.")
       end
 
-      it "expects that person is a child of the tenant" do
-        record.age2 = 14
+      it "validates person over 19 is not child" do
+        record.age2 = 20
         record.relat2 = "C"
-        household_validator.validate_household_number_of_other_members(record)
-        expect(record.errors["relat2"]).to be_empty
-        expect(record.errors["age2"]).to be_empty
+        household_validator.validate_person_age_matches_relationship(record)
+        expect(record.errors["age2"])
+          .to include("Age must be 19 or under as you told us person 2's relationship to the lead tenant is child")
+        expect(record.errors["relat2"])
+          .not_to include("Answer cannot be child, as you told us person 2 is over 19")
       end
+    end
+  end
 
+  describe "#validate_person_age_matches_economic_status" do
+    context "when the household contains a person under 16" do
```

Comment:
Does this need an explicit 2023 context?

Reply:
Yup, I think so! Thanks!